### PR TITLE
Hot fix for literals & identifier extractors:

### PR DESCRIPTION
**Pull request by EgorBu**
* replace bblfsh.filter with simple traversal algorithm
* add traversal algorithm to utils
* add bblfsh roles

Signed-off-by: egor <egor@sourced.tech>

### CODE REVIEW

File: sourced/ml/algorithms/uast_ids_to_bag.py [hunk 1]
```diff
@@ -70,3 +72,23 @@ def __init__(self, token2index=None, token_parser=None):
         """
         token_parser = TokenParser() if token_parser is None else token_parser
         super().__init__(token2index, token_parser)
+
```
Comment (vmarkovtsev) [inline sourced/ml/algorithms/uast_ids_to_bag.py]: As Konst found, `roleQualified` should be removed

File: sourced/ml/algorithms/uast_ids_to_bag.py [hunk 2]
```diff
@@ -70,3 +72,23 @@ def __init__(self, token2index=None, token_parser=None):
...
+        """
+        Converts a UAST to a weighed bag-of-identifiers. The weights are identifiers frequencies.
+        The tokens are preprocessed by _token_parser.
+        Overwrite __call__ to avoid issues with `bblfsh.filter`.
+
+        :param uast: The UAST root node.
+        :return: bag
+        """
+        nodes = [node for node in uast2sequence(uast)
+                 if IDENTIFIER in node.roles and QUALIFIED not in node.roles]
```
Comment (vmarkovtsev) [inline sourced/ml/algorithms/uast_ids_to_bag.py]: ~QUALIFIED~

File: sourced/ml/algorithms/uast_ids_to_bag.py [hunk 3]
```diff
@@ -70,3 +72,23 @@ def __init__(self, token2index=None, token_parser=None):
         """
         token_parser = TokenParser() if token_parser is None else token_parser
         super().__init__(token2index, token_parser)
+
+    def __call__(self, uast):
+        """
+        Converts a UAST to a weighed bag-of-identifiers. The weights are identifiers frequencies.
```
Comment (vmarkovtsev) [inline sourced/ml/algorithms/uast_ids_to_bag.py]: Please add a big visible FIXME for libuast override

File: sourced/ml/algorithms/uast_ids_to_bag.py [hunk 4]
```diff
@@ -4,6 +4,8 @@
 
 from sourced.ml.algorithms import TokenParser, NoopTokenParser
 from sourced.ml.algorithms.uast_to_bag import Uast2BagBase
+from sourced.ml.utils import IDENTIFIER, QUALIFIED
+from sourced.ml.utils import uast2sequence
```
Comment (vmarkovtsev) [inline sourced/ml/algorithms/uast_ids_to_bag.py]: `uast2sequence` should be somewhere in "algorithms"

File: sourced/ml/extractors/literals.py [hunk 5]
```diff
@@ -27,6 +29,25 @@ def __init__(self, token2index=None, token_parser=None):
         token_parser = HashedTokenParser() if token_parser is None else token_parser
         super().__init__(token2index, token_parser)
 
+    def __call__(self, uast):
+        """
+        Converts a UAST to a weighed bag-of-literals. The weights are literals frequencies.
```
Comment (vmarkovtsev) [inline sourced/ml/extractors/literals.py]: Same big FIXME here

File: sourced/ml/algorithms/uast_ids_to_bag.py (hunk 1 above)
Comment (EgorBu) [reply sourced/ml/algorithms/uast_ids_to_bag.py]: yes, checked it in https://github.com/src-d/ml/pull/207 - will be removed

File: sourced/ml/algorithms/uast_ids_to_bag.py (hunk 3 above)
Comment (EgorBu) [reply sourced/ml/algorithms/uast_ids_to_bag.py]: I will put `FIXME bblfsh.filter` - I'm not sure if it's `bblfsh-client` problem or `libuast`. Anyway I submitted issue to `bblfsh-client`:  https://github.com/bblfsh/client-python/issues/92

File: sourced/ml/algorithms/uast_ids_to_bag.py (hunk 4 above)
Comment (EgorBu) [reply sourced/ml/algorithms/uast_ids_to_bag.py]: Will move it

**Comment by vmarkovtsev** [comment]
@EgorBu Travis is broken

File: sourced/ml/utils/__init__.py [hunk 6]
```diff
@@ -1,4 +1,5 @@
 from sourced.ml.utils.bigartm import install_bigartm
+from sourced.ml.utils.bblfsh_roles import IDENTIFIER, QUALIFIED, LITERAL
```
Comment (zurk) [inline sourced/ml/utils/__init__.py]: wow. 
Can we avoid doing like that?
It is hard to understand what it is about if you see in code just alone `LITERAL`.

Usually I prefer to import like:
```python 
from sourced.ml.utils import bblfsh_roles

# And use in code like: 
bblfsh_roles.LITERAL #  Here you see what it is exactly.
```

Comment (zurk) [reply sourced/ml/utils/__init__.py]: ^

Comment (vmarkovtsev) [reply sourced/ml/utils/__init__.py]: Konst is saying sensible thing, makes sense to fix in the future

**Pull request merged**